docs/methodology.md

```diff
@@ -0,0 +1,294 @@
+
+&ensp; &ensp; $f'(\mu)=(b+2c\cdot \mu+3d\cdot \mu^2)/(x_1-x_0)$
+
+Evaluate at 0 and 1:
+
+&ensp; &ensp; $f(0)=a$
+     
+&ensp; &ensp; $f(1)=a+b+c+d$
+     
+&ensp; &ensp; $f'(0)=b/(x_1-x_0)$
+     
+&ensp; &ensp; $f'(1)=(b+2c+3d)/(x_1-x_0)$
+     
+
+We can invert this system of equations to find the polynomial coefficients:[^1]
+
+$$
+\begin{align*}
+&a = &&f(0)   	&&   		&&&   							&&\\
+&b = && 		&&			&(x_1-x_0)\cdot &f'(0) 		&&\\ 
+&c=  &-3&f(0) 	&+3&f(1) 	&-2(x_1-x_0)\cdot &f'(0)	 	&-(x_1-x_0)\cdot &f'(1)\\    
+&d = &2&f(0) 	&-2&f(1) 	&+(x_1-x_0)\cdot &f'(0)			&+(x_1-x_0)\cdot &f'(1)\\
+\end{align*}
+$$
+     
+Substitute into $f(\mu)$ and arrange the coefficient sums in descending powers of $\mu$:
+
+$$
+\begin{align*}
+f(\mu)=
+\quad&(2\mu^3-3\mu^2+1)&\cdot\quad &f(0)\\
++&(-2\mu^3+3\mu^2)&\cdot\quad &f(1)\\
++&(\mu^3-2\mu^2+\mu)\cdot (x_1-x_0)&\cdot\quad  &f'(0)\\
++&(\mu^3-\mu^2)\cdot (x_1-x_0)&\cdot\quad &f'(1)\\
+\end{align*}
+$$
+
+### Slope estimation
+Whereas we know the function values, $f(0)$ and $f(1)$, we can only estimate the slopes, $f'(0)$ and $f'(1)$, from the data, itself. Several methods for estimating these slopes have been proposed.[^2] A common assumption is that the slope at any grid point is equal to that of the line containing the previous and next points on the axis:
+
+&ensp; &ensp; $f'(0)=(f(1)-f(-1))/(x_1-x_{-1})$
+
+&ensp; &ensp; $f'(1)=(f(2)-f(0))/(x_2-x_0)$
+
+<p align="center">
+    <img width="60%" src="./figs/fig_slope_est.png"> 
+</p>
+
+Now we have
+
+$$
+\begin{align*}
+f(\mu)=
+\quad&(2\mu^3-3\mu^2+1)&\cdot\quad &f(0)\\
++&(-2\mu^3+3\mu^2)&\cdot\quad &f(1)\\
++&(\mu^3-2\mu^2+\mu)\cdot (x_1-x_0)/(x_1-x_{-1})&\cdot\quad  &(f(1)-f(-1))\\
++&(\mu^3-\mu^2)\cdot (x_1-x_0)/(x_2-x_0)&\cdot\quad &(f(2)-f(0))\\
+\end{align*}
+$$
+
+In the cubic case, the `interpolation_coefficients` have the following forms:
+
+&ensp; &ensp; $C_0=2\mu^3-3\mu^2+1, \quad  C_1=-2\mu^3+3\mu^2$
+
+The following two variables are labeled in **btwxt** source code as `cubic_slope_coefficients`.
+
+&ensp; &ensp; $D_0=(\mu^3-2\mu^2+\mu)\cdot S_0, \quad D_1=(\mu^3-\mu^2)\cdot S_1$
+
+The ratios
+
+&ensp; &ensp; $S_0=(x_1-x_0)/(x_1-x_{-1}), \quad S_1=(x_1-x_0)/(x_2-x_0)$
+
+are assigned to variables in **btwxt** labeled `cubic_spacing_ratios`. 
+
+We can rewrite the polynomial:   
+
+$$
+\begin{align*}
+f(\mu)=
+\quad &C_0&\cdot\quad &f(0)\\
++&C_1&\cdot\quad &f(1)\\
++&D_0&\cdot\quad &(f(1)-f(-1))\\
++&D_1&\cdot\quad &(f(2)-f(0))
+\end{align*}
+$$
+
+Finally, identify the coefficients for the individual function values:
+
+$$
+\begin{align*}
+f(\mu)=
+-&D_0&\cdot\quad &f(-1)\\
++&(C_0 - D_1)&\cdot \quad &f(0)\\
++&(C_1+D_0)&\cdot\quad &f(1)\\
++&D_1&\cdot\quad &f(2)
+\end{align*}
+$$
+
+For cubic interpolation, the `weighting factors` are
+
+&ensp; &ensp; $W_{-1}=-D_0, \quad  W_0=C_0-D_1, \quad W_1=C_1+D_0, \quad W_2=D_1$
+
+
+<p align="center">
+    <img width="60%" src="./figs/fig_cubic.png"> 
+</p>
```

Comment:
This still doesn't look like it's matching the approximated slopes from the previous figure.